engines/vcruise/vcruise.cpp

```diff
@@ -0,0 +1,187 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "engines/util.h"
+
+#include "common/config-manager.h"
+#include "common/events.h"
+#include "common/system.h"
+#include "common/algorithm.h"
+
+#include "vcruise/runtime.h"
+#include "vcruise/vcruise.h"
+
+namespace VCruise {
+
+VCruiseEngine::VCruiseEngine(OSystem *syst, const VCruiseGameDescription *gameDesc) : Engine(syst), _gameDescription(gameDesc) {
+	const Common::FSNode gameDataDir(ConfMan.get("path"));
+}
+
+VCruiseEngine::~VCruiseEngine() {
+}
+
+void VCruiseEngine::handleEvents() {
+	Common::Event evt;
+	Common::EventManager *eventMan = _system->getEventManager();
+
+	while (eventMan->pollEvent(evt)) {
+		switch (evt.type) {
+		case Common::EVENT_LBUTTONDOWN:
+			_runtime->onLButtonDown(evt.mouse.x, evt.mouse.y);
+			break;
+		case Common::EVENT_LBUTTONUP:
+			_runtime->onLButtonUp(evt.mouse.x, evt.mouse.y);
+			break;
+		case Common::EVENT_MOUSEMOVE:
+			_runtime->onMouseMove(evt.mouse.x, evt.mouse.y);
+			break;
+		case Common::EVENT_KEYDOWN:
+			_runtime->onKeyDown(evt.kbd.keycode);
+			break;
+		default:
+			break;
+		}
+	}
+}
+
+Common::Error VCruiseEngine::run() {
+	Common::List<Graphics::PixelFormat> pixelFormats = _system->getSupportedFormats();
```

Comment:
I'd recommend using `Engine::initGraphicsModes(const Graphics::ModeList &modes)` instead of this whole construct

Reply:
`initGraphicsModes` is just for sizes, and I don't like sending a mode list because `findCompatibleFormat` requires exact mode matches when really I don't care about channel order or alpha channel presence as long as a mode with enough RGB bits is available.